cps_data/pycps/impute.py

```diff
@@ -0,0 +1,262 @@
+    )
+    # cap family size at 5 to match with the PUF
+    # data["fam_size"] = np.minimum(data["XTOT"], 5)
+    data["agede"] = (
+        (data["age_head"] >=
+         FILINGPARAMS.elderly_age[CPS_YR_IDX]).astype(int) +
+        (data["age_spouse"] >=
+         FILINGPARAMS.elderly_age[CPS_YR_IDX]).astype(int)
+    )
+    data["constant"] = np.ones(len(data))
+
+    np.random.seed(5410)  # set random seed before imputations
+
+    # calculate imputations
+    # dictionary to hold parameters used in logit/ols imputations
+    IMPUTATION_PARAMS = {
+        "CGAGIX": {
+            "x_vars": [
+                "lntot_inc", "joint_filer", "fam_size", "lninterest",
+                "lndivs", "lne01500", "constant"
+            ],
+            "logit_betas": logit_betas["cg_logit"],
+            "ols_betas": ols_betas["cg_ols"],
+            "l_adj": 0.065,
+            "o_adj": 1.95,
+            "prob_mult": 1.
+        },
+        "TIRAD": {
+            "x_vars": X_VARS,
+            "logit_betas": logit_betas["ira_logit"],
+            "ols_betas": ols_betas["ira_ols"],
+            "l_adj": 0.,
+            "o_adj": 1.7,
+            "prob_mult": 1.25
+        },
+        "ADJIRA": {
+            "x_vars": X_VARS,
+            "logit_betas": logit_betas["irac_logit"],
+            "ols_betas": ols_betas["irac_ols"],
+            "l_adj": 0.,
+            "o_adj": 1.,
+            "prob_mult": .06
+        },
+        "KEOGH": {
+            "x_vars": X_VARS,
+            "logit_betas": logit_betas["sep_logit"],
+            "ols_betas": ols_betas["sep_ols"],
+            "l_adj": 0.,
+            "o_adj": 1.,
+            "prob_mult": 1.
+        },
+        "SEHEALTH": {
+            "x_vars": X_VARS,
+            "logit_betas": logit_betas["sehi_logit"],
+            "ols_betas": ols_betas["sehi_ols"],
+            "l_adj": 0.,
+            "o_adj": 1.,
+            "prob_mult": 1.6
+        },
+        "SLINT": {
+            "x_vars": X_VARS,
+            "logit_betas": logit_betas["sl_logit"],
+            "ols_betas": ols_betas["sl_ols"],
+            "l_adj": 0.,
+            "o_adj": 1.,
+            "prob_mult": 1.1
+        },
+        "CDC": {
+            "x_vars": X_VARS,
+            "logit_betas": logit_betas["cdc_logit"],
+            "ols_betas": ols_betas["cdc_ols"],
+            "l_adj": 0.,
+            "o_adj": 1.,
+            "prob_mult": 1.
+        },
+        "MEDEX": {
+            "x_vars": X_VARS,
+            "logit_betas": logit_betas["medex_logit"],
+            "ols_betas": ols_betas["medex_ols"],
+            "l_adj": 0.,
+            "o_adj": 1.,
+            "prob_mult": 1.
+        }
+    }
+
+    for var, params in IMPUTATION_PARAMS.items():
+        data[var] = impute(data, **params)
+
+    # tobit models
+    TOBIT_XVARS = [
+        "lntot_inc", "joint_filer", "fam_size", "agede", "constant"
+    ]
+
+    data["CHARITABLE"] = tobit(
+        data, ols_betas["char_ols"], TOBIT_XVARS, 48765.45, 1.
```

Comment:
Where this 48765.45 from?

And in general there are a lot of hard-coded numbers in this file. Are they documented? Ideally I think these would be their own defined constants rather than entered in functions, or even better would be loading them from CSVs.

Reply:
All of the hard coded numbers come from John and can be found in [this](https://github.com/PSLmodels/taxdata/blob/master/cps_data/State%20Database%20-%202015/Production%20File/Imputations/Impute3TobitV4.sas) file. I carried them over to be consistent with what we've been doing for imputations up to this point. Agreed that it's less than ideal. I'd like for one of the next TaxData projects to be revisiting all of the imputations we do on the CPS to make them more transparent, reproducible, easier to update, etc. Getting rid of all the hard coded numbers would be part of that work.